theme-kit/env.ts

```diff
@@ -0,0 +1,80 @@
+/* eslint-disable @typescript-eslint/no-use-before-define */
+import { headers } from 'next/headers';
+import parseDataUrl from 'parse-data-url';
+import { z, ZodError } from 'zod';
+
+import { assertServerEnv } from '@/utils';
+
+const Schema = z
+    .object({
+        PREZLY_ACCESS_TOKEN: z.string(),
+        PREZLY_NEWSROOM_UUID: z.string(),
+        PREZLY_THEME_UUID: z.string().optional(),
+        PREZLY_API_BASEURL: z.string().optional(),
+
+        ALGOLIA_API_KEY: z.string(),
+        ALGOLIA_APP_ID: z.string(),
+        ALGOLIA_INDEX: z.string(),
+    })
+    .catchall(z.string());
+
+type ExpectedEnv = z.infer<typeof Schema>;
+
+export function env(): ExpectedEnv {
+    assertServerEnv('env');
+
+    return validateEnv(getEnvVariables(process.env, headers()));
+}
```

Comment:
This helper inspects `process.env` and `headers()` for relevant values and validates it against the `zod` schema, making sure we don't have any misconfiguration and that the returned object always complies with the expected interface